rcgen/src/lib.rs

```diff
@@ -1562,49 +1555,31 @@ impl Certificate {
 		pem::encode_config(&Pem::new("CERTIFICATE", self.der()), ENCODE_CONFIG)
 	}
 	/// Serializes a certificate signing request in binary DER format
-	pub fn serialize_request_der(&self) -> Result<Vec<u8>, Error> {
+	pub fn serialize_request_der(&self, subject_key: &KeyPair) -> Result<Vec<u8>, Error> {
```

Comment:
Might be helpful to add some clarification to the docstrings for `serialize_request_*()` about what `subject_key` and how it's used? I can't think of a better name but also feel like this name is maybe not the most obvious.

Reply:
I added more clarification here in a separate commit. I also explicitly called out that this API still has the footgun we're fixing in this branch for the `Certificate` type (e.g. calling `serialize_request_der` and then `serialize_request_pem` from the same `Certificate` isn't going to do what you expect).